app/master/cluster_master.py

```diff
@@ -62,12 +65,24 @@ def api_representation(self):
             'slaves': slaves_representation,
         }
 
-    def builds(self):
+    def builds(self, offset: int=None, limit: int=None) -> List['Build']:
         """
-        Returns a list of all builds
-        :rtype: list[Build]
+        Returns a list of all builds. Both offset and limit are assumed to be nonnegative at this point
+        since the corrections or negative offsets and limits are done when the query parameters are first initialized.
```

Comment:
1. I like the idea of adding an extra check, especially since it's so easy to do.
`offset = max(offset or 0, 0)`
In my perfect world every method would be callable with every combination of arguments that its type signature permits. Since this method's type signature allows for negative arguments, it should behave sanely when negative arguments are passed in. (Sometimes obeying this philosophy is more trouble than it's worth, but in this case it's a one liner)

2. iirc @josephharrington suggested allowing you to pass in negative offsets in order to get the most recent builds. So if I passed in `offset=0&limit=-1` it would give me the most recent build. I liked that suggestion. Could we rename the variables to something like `start` and `range` and then do
```python
start = start % num_builds
end = min(start + range,  num_builds)
end = end % num_builds
start, end = min(start, end), max(start, end)
return [self._all_builds_by_id[key] for key in islice(self._all_build_by_id, start, end)]
```